runtime/arc.js

```diff
@@ -102,11 +102,12 @@ class Arc {
     return [...this._viewsById.values()];
   }
 
-  clone() {
+  // Makes a copy of the arc used for speculative execution.
+  makeSpeculative() {
```

Comment:
I think makeSpeculativeClone is fine, it's only used in one place.

cloneForSpeculativeExecution would be another option?